xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp

```diff
@@ -776,9 +778,9 @@ void CVideoPlayerVideo::Flush(bool sync)
   /* flush using message as this get's called from VideoPlayer thread */
   /* and any demux packet that has been taken out of queue need to */
   /* be disposed of before we flush */
-  FlushMessages();
   SendMessage(new CDVDMsgBool(CDVDMsg::GENERAL_FLUSH, sync), 1);
   m_bAbortOutput = true;
+  m_flushEvent.Wait();
```

Comment:
there is no need to wait here and bypassing the method VP provides for sync. VP syncs with streamplayers by sending a CDVDMsgGeneralSynchronize msg. This is done in CVideoPlayer::FlushBuffers

Reply:
Thx for looking at it @FernetMenta , Do I understand your comment right that no new packet is send from VP to VPV while VPV thread is waiting for the FLUSH message to be processed?

Reply:
yes: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L3961

Reply:
Thx!